### ISSUE TITLE
Refactor AM modules to use chiventure context struct

### ISSUE THREAD

**Issue by MTUC08** [uchicago-cs/chiventure#250]
Refactor action management functions to use the chiventure context struct. This allows for simpler passing of arguments between group functions and is anticipated to be the norm for sprint 4. This includes testing functions as well.

Additionally, CLI is being added to this issue so that they can refactor operations.c with our new parameters.

**Comment by SnowballAntrobus** [comment]
I can help with the testing functions.

**Comment by MTUC08** [comment]
So my current line of logic is that since do_path_action is the only one that takes in/uses a game_t struct, it's the only one that needs to be refactored to use a chiventure context struct.

**Comment by MTUC08** [comment]
Update: Just refactored do_path_action c and h to take in and handle the chiventure context struct instead of a game struct. This was done on branch actions/chiventure_ctx_t and after Dante refactors the corresponding tests, will be merged into actions/do_path_191 so it can go into actions/dev with issue #302 .

**Comment by SnowballAntrobus** [comment]
I have added a member of CLI so that they can refactor their code to our new function parameters.

**Comment by ecoble** [comment]
UPDATE: Changes CLI's use of action management's do_item_action, do_path_action, and do_item_item_action to not pass the game struct as a parameter (for do_item_action and do_item_item_action), or replacing the game struct parameter with a chiventure context struct parameter (for do_path_action)

**Comment by SnowballAntrobus** [comment]
We had an issue with a circular dependency where ctx.h was calling cmd.h which was calling actionmanagement.h which was calling ctx.h. I think we have now resolved this with help from CLI.

**Comment by SnowballAntrobus** [comment]
Ok we should be done with the refactor both of tests and actionmanagment.c/.h using the new chiventure context struct. The main issue was fixing makefiles because now we have to link with cli and ui since they are used in the struct. We will now be making a PR into actions/do_path_191.

**Comment by MTUC08** [comment]
In this issue, we changed the game_t input parameter to the do_path function to the shared chiventure context struct chiventure_ctx_t and removed the game_t input parameter from the other do functions as they were not being used to actually implement the function. We then refactored the do_path_function and tests from issue #191 and #193 to reflect the new refactorization. CLI also refactored their function calls based on this change. Finally, we fixed the makefiles to link to the cli, ui, and common libraries to reflect the new chiventure context struct. 

Looking forward, the files from this issue will be PRed to do_path_191 in #321 and then PRed to actions/dev in #302 where it will be joined by the other sprint 4 refactored (i.e. with conditions/attributes) do functions in a final PR to master.
